uni/udb/breakpoint.icn

```diff
@@ -76,75 +75,104 @@ end
+   bpTable["type"] := "breakpoint"
 
-   if \verbose then 
+   if \verbose then {
       src_code ||:= "\n     " || bp.line || ": " || bp.code
+      bpTable["code"] := bp.code
+   }
 
-   if \bp.coexp then
-      coexp ||:= " coexp " || bp.coexp.id 
+   if \bp.coexp then {
+      coexp ||:= " coexp " || bp.coexp.id
+      bpTable["coexp"] := bp.coexp.id
+   }
 
-   return breakpoint || coexp || src_code
+   bpTable["consoleMsg"] := breakpoint || coexp || src_code
+   return bpTable
 end
 
 #
 # prints out a list of the existing breakpoints used by cmdBreak
 #
 method printBreakInfo(cmd)
    local x, L, bp, name
+   local resultTable := table("cmd", cmd, "breakpoints", list())
 
    if type(cmd) == "list" then name := cmd[3]
    msg := ""
    if /name then {
       every x := key(breakPoints) do {
-	 L := breakPoints[x]
-	 every bp := !L do {
-            msg ||:= constructBpMsg(bp)
-            }
-	 }
+         L := breakPoints[x]
+         every bp := !L do {
+            bpTable := constructBpMsg(bp)
+            msg ||:= bpTable["consoleMsg"]
+            push(resultTable["breakpoints"], bpTable)
+         }
       }
+   }
    else {
       if integer(name) then {
-	 every x := key(breakPoints) do {
-	    L := breakPoints[x]
-	    every bp := !L do {
-	       if name = bp.id then {
-                  msg := constructBpMsg(bp)
-	          break break
-	          }
-	       }
-	    msg||:="\n   Breakpoint #"||name||" is not available."
-	    }
-	 }
+	      every x := key(breakPoints) do {
+            L := breakPoints[x]
+            every bp := !L do {
+               if name = bp.id then {
+                  bpTable := constructBpMsg(bp)
+                  msg ||:= bpTable["consoleMsg"]
+                  push(resultTable["breakpoints"], bpTable)
+                  break break
+               }
+            }
+            msg||:="\n   Breakpoint #"||name||" is not available."
+            push(resultTable["breakpoints"], bpTable)
+	      }
+	   }
       else {
-	 if not find(".icn",name) then name ||:= ".icn"
-	 if DState.srcFile.isSrcFile(name) then {
-	    if L:=breakPoints[name] & type(L) == "list" then {
-	       every bp := !L do
-                  msg ||:= constructBpMsg(bp)
-	       }
-	    else
-	       msg := "\n   No breakpoints available in "||name
-	    }
-	 else
-	    msg := "\n   Source file \""||name||"\" is not in the binary !!!"
-	 }
+         if not find(".icn",name) then name ||:= ".icn"
+         if DState.srcFile.isSrcFile(name) then {
+            if L:=breakPoints[name] & type(L) == "list" then {
+               every bp := !L do {
+                  bpTable := constructBpMsg(bp)
+                  msg ||:= bpTable["consoleMsg"]
+                  push(resultTable["breakpoints"], bpTable)
+               }
+            }
+            else {
+               msg := "\n   No breakpoints available in "||name
+               setResultTable(resultTable, "No breakpoints available in " || name, "__true__")
```

Comment:
I know I suggested to make the `setResultTable()` function, but maybe consider use the table syntax instead?

Reply:
There are 52 instances of `setResultTable()` used in breakpoint. With it, we get the option to change table keys in one place instead of having to change them in all 52 instances. The table syntax does have the pro of looking prettier, but I think `setResultTable()` may be more practical.

Reply:
agreed.